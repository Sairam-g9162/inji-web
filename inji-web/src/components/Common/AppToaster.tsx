import React from "react";
import { ToastContainer, toast } from "react-toastify";
import { useSelector } from "react-redux";
import { RootState } from "../../types/redux";
import { isRTL } from "../../utils/i18n";

const CustomToast = ({ message }: { message: string }) => (
    <div data-testid="custom-toast">{message}</div>
);

export const AppToaster: React.FC = () => {
    const language = useSelector((state: RootState) => state.common.language);
    const positionClass = isRTL(language) ? "top-left" : "top-right";

    // Example of triggering a toast
    React.useEffect(() => {
        toast(<CustomToast message="Test message" />);
    }, []);

    return (
        <div className={`toast-container ${positionClass}`} data-testid="toast-container">
            <ToastContainer
                position={isRTL(language) ? "top-left" : "top-right"}
                autoClose={1000}
                hideProgressBar
                newestOnTop
                closeOnClick
                rtl={isRTL(language)}
                icon={<React.Fragment />}
                pauseOnFocusLoss
                draggable
                pauseOnHover
                style={{ width: '400px' }}
                theme="colored"
            />
        </div>
    );
};
