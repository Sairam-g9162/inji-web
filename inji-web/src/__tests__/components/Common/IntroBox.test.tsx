import React from 'react';
import { render, screen } from '@testing-library/react';
import {IntroBox} from "../../../components/Common/IntroBox";

describe("Test Intro Box Layout",() => {
    test('check the presence of the container', () => {
        const {asFragment} = render(<IntroBox/>);
        expect(asFragment()).toMatchSnapshot();
    });
})


describe("Test Intro Box Content",() => {
    test('check if content is rendered properly', () => {
        render(<IntroBox />);
        const headerElement = screen.getByTestId("IntroBox-Text");
        expect(headerElement).toHaveTextContent("Intro.title")
    });
    test('check if content is rendered properly subTitle', () => {
        render(<IntroBox />);
        const headerElement = screen.getByTestId("IntroBox-SubText");
        expect(headerElement).toHaveTextContent("Intro.subTitle")
    });
})
