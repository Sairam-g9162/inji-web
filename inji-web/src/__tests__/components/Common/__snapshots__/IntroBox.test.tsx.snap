// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Intro Box Layout check the presence of the container 1`] = `
<DocumentFragment>
  <div
    class="text-center pt-20 pb-10"
    data-testid="IntroBox-Container"
  >
    <h2
      class="text-3xl text-iw-title "
      data-testid="IntroBox-Text"
    >
      Intro.title
    </h2>
    <p
      class="mt-2 text-lg text-iw-subTitle"
      data-testid="IntroBox-SubText"
    >
      Intro.subTitle
    </p>
  </div>
</DocumentFragment>
`;
